proposals/20240606-jwt-handling.md

```diff
@@ -0,0 +1,125 @@
+# Standardise JWTs Usage
+
+**Authors**: Kimonas Sotirchos @kimwnasptd
+
+## Requirements
+- OIDC logic, via [Istio external authoriser](https://istio.io/latest/docs/tasks/security/authorization/authz-custom/), should be adding `id_token` to http requests in `Authorization: Bearer <token>` headers
+## Scope
+This proposal aims to standardise how the Kubeflow backends should be handling the user information (name of user, groups they belong to), living in JWTs and http headers.
+
+This proposal takes as a requirement that users should be able to use K8s Tokens as `Authorization: Bearer <token>` headers in their request from inside the cluster.
+### In
+- Which component should be validating JWTs (id-tokens from OIDC or K8s ServiceAccount tokens)
+- Define where the backends should expect to find user related information
+- Define how different token "types" (dex, k8s) should be handled
```

Comment:
Ah good point to clarify, I meant about different JWT **issuers**. At the end of the day what the token represents (user, k8s-service account, some vm) should be irrelevant to the apps.

Adding a small clarifications. I'll leave the comment open though in case you had further suggestions here
5a0b4e27

Reply:
I like the updated version. Thanks.